### PR TITLE
Fix getTitle to return the actual title

### LINKED ISSUE #584: `SeachIterator::getTitle()` gives unaccented title
`SearchIterator::getTitle()` gives the unaccented title instead of the actual title. That is, even if the title is `"DeLorean"`, we get `"delorean"` as the output.

When we index the title, the value stored in `title:0` slot of the `valuesmap` is the unaccented title. This happens because `zim::removeAccents()` is called in the constructor of `DefaultIndexData`.  

This was missed by our unit tests for several reasons like calling the `getTitle` from the dereferenced `entry`, non-availability of mix of upper and lower case in tests where we actually call it from the search iterator. An easy fix is to drop this behavior from the constructor because we are anyway explicitly calling `zim::removeAccents()` where it is really required in `XapianIndexer::indexTitle()`. Suggestions?

### PR DESCRIPTION
**Pull request by maneeshpm**
Fixes #584 
SearchIterator::getTitle() should return the actual title of the entry rather than returning it from the valuesmap which is unaccented.

### CODE REVIEW

**Comment by kelson42** [comment]
@maneeshpm Please extend the test set with accented and cased tests.

**Comment by maneeshpm** [comment]
@kelson42 Added it to one of the existing tests :smile:

**Comment by kelson42** [comment]
@maneeshpm Please add a test with accents

File: src/search_internal.h [hunk 1]
```diff
@@ -125,7 +125,7 @@ struct SearchIterator::InternalData {
     }
 
     Entry& get_entry() {
-        if ( !_entry ) {
+        if (!_entry || !document_fetched) {
```
Comment (mgautierfr) [inline src/search_internal.h:128]: Why adding this test ?
Can we have a entry but no document_fetched ? (knowing that we build the entry from the document)

Comment (maneeshpm) [reply src/search_internal.h:128]: Aah my bad, I must have added it in some testing and forgot to remove it. I'll fix it back quickly.

**Review by mgautierfr — APPROVED** [review]

**Pull request merged**